### PR TITLE
did the rest of binsearch

### PR DESCRIPTION
**Pull request by SansPapyrus683**
i'll get around to adding python support for the last problem

### CODE REVIEW

File: content/3_Silver/Binary_Search.mdx [hunk 1]
```diff
@@ -469,69 +480,49 @@ Here we write the check function as a
...
+	static int n;
+	static int k;
+	static List<Integer> a = new ArrayList<>();
 	public static void main(String[] args){
 		Kattio io = new Kattio();
 
-		n = io.nextInt(); k = io.nextInt();
-		for(int i = 0; i < n; i++){
+		n = io.nextInt();
+        k = io.nextInt();
```
Comment (maggie-j-liu) [inline content/3_Silver/Binary_Search.mdx:491]: indentation is messed up here

File: content/3_Silver/Binary_Search.mdx [hunk 2]
```diff
@@ -469,69 +480,49 @@ Here we write the check function as a
...
+		for (int i = 0; i < n; i++) {
 			int t = io.nextInt();
-			v.add(t);
+			a.add(t);
 		}
-		Collections.sort(v);
-		io.println(lastTrue(1, (int)2e9));
+		Collections.sort(a);
+
+        // Use 2e9 instead of Integer.MAX_VALUE to avoid overflow
```
Comment (maggie-j-liu) [inline content/3_Silver/Binary_Search.mdx:498]: and here

File: content/3_Silver/Binary_Search.mdx [hunk 3]
```diff
@@ -894,37 +924,31 @@ functions.
...
+	for (int i = 0; i < N; i++) {
+		cin >> v[i];
+	}
+	
+	sort(begin(v), end(v));
+	for (int i = 0; i < Q; i++) {
+		int A;
+		int B;
+		cin >> A >> B;
+		cout << upper_bound(begin(v), end(v), B) - lower_bound(begin(v), end(v), A) << "\n";
```
Comment (maggie-j-liu) [inline content/3_Silver/Binary_Search.mdx:951]: this line is too long

Comment (SansPapyrus683) [reply content/3_Silver/Binary_Search.mdx:951]: 89 chars, should be fine

Comment (maggie-j-liu) [reply content/3_Silver/Binary_Search.mdx:951]: uhh it's 92 characters, and since the suggested is 80 that's too long

Comment (SansPapyrus683) [reply content/3_Silver/Binary_Search.mdx:951]: ok cool
how should i split it up
all the previous methods i tried made it look really weird

Comment (maggie-j-liu) [reply content/3_Silver/Binary_Search.mdx:951]: does this look okay?
```cpp
		cout << upper_bound(begin(v), end(v), B) 
					- lower_bound(begin(v), end(v), A) << "\n";
```

Comment (SansPapyrus683) [reply content/3_Silver/Binary_Search.mdx:951]: looks a bit sus
but better than nothing ig

**Review by maggie-j-liu — APPROVED** [review]

**Review by nathangong — APPROVED** [review]
Java changes look good

File: content/3_Silver/Binary_Search.mdx [hunk 4]
```diff
@@ -540,34 +531,36 @@ public class BinarySearch{
...
-	operationsNeeded = 0
-	for i in range((n-1)//2,n):
-		operationsNeeded += max(0, x-v[i])
-	if operationsNeeded <= k:
-		return True
-	else:
-		return False
+from typing import Callable
+
+
```
Comment (ryanchou-dev) [inline content/3_Silver/Binary_Search.mdx:536]:
```suggestion
```

File: content/3_Silver/Binary_Search.mdx [hunk 5]
```diff
@@ -790,14 +817,17 @@ public static int firstTrue(int lo, int hi, Predicate<Integer> f) {
 <PySection>
 
 ```python
-def firstTrue(lo, hi, f):
-	hi+=1
+from typing import Callable
+
+
```
Comment (ryanchou-dev) [inline content/3_Silver/Binary_Search.mdx:822]:
```suggestion
```

**Review by ryanchou-dev — APPROVED** [review]
python and c++ looks good

File: content/3_Silver/Binary_Search.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply content/3_Silver/Binary_Search.mdx:536]: ryan did you not see my comments on the previous pr

**Comment by SansPapyrus683** [comment]
@nathangong could you possibly review again
i did some pretty big changes lol

**Review by maggie-j-liu — APPROVED** [review]

**Review by nathangong — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
ok uh can we merge now
or do i need like 1 more approval

**Review by brad182 — APPROVED** [review]

**Pull request merged**